### DISCUSSION TITLE
[openai] Error: argument should be a bytes-like object or ASCII string, not 'list'General Discussion

### DISCUSSION THREAD

**Discussion by shanaozi** [langgenius/dify#6805] [General]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### Content

[openai] Error: argument should be a bytes-like object or ASCII string, not 'list'一直无法调用openai的向量化接口 无法dify配置 求求了

**Comment by crazywoola** [comment]
是 OpenAI 官方的 key 吗

**Reply by shanaozi** [reply]
是站点的,必须要官方的吗,但是其他接口都可以 除了向量化这个接口就报错

**Reply by shanaozi** [reply]
pi-1         |   File "/app/api/controllers/console/setup.py", line 75, in decorated
api-1         |     return view(*args, **kwargs)
api-1         |   File "/app/api/libs/login.py", line 91, in decorated_view
nginx-1       | 172.21.0.1 - - [31/Jul/2024:01:16:45 +0000] "POST /console/api/datasets/7cf35447-72f8-4fec-8bb0-d5e82c4b9111/hit-testing HTTP/1.1" 500 146 "http://localhost:82/datasets/7cf35447-72f8-4fec-8bb0-d5e82c4b9111/hitTesting" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/127.0.0.0 Safari/537.36" "-"
api-1         |     return current_app.ensure_sync(func)(*args, **kwargs)
api-1         |   File "/app/api/controllers/console/wraps.py", line 22, in decorated
api-1         |     return view(*args, **kwargs)
api-1         |   File "/app/api/controllers/console/datasets/hit_testing.py", line 83, in post
api-1         |     raise InternalServerError(str(e))
api-1         | werkzeug.exceptions.InternalServerError: 500 Internal Server Error: [openai] Error: argument should be a bytes-like object or ASCII string, not 'list'
nginx-1       | 172.21.0.1 - - [31/Jul/2024:01:16:47 +0000] "GET /console/api/datasets/2826b002-6ad9-40ab-b4ed-e1db06086f40/batch/20240731011039466024/indexing-status HTTP/1.1" 200 335 "http://localhost:82/datasets/7cf35447-72f8-4fec-8bb0-d5e82c4b9111/hitTesting" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/127.0.0.0 Safari/537.36" "-"
nginx-1       | 172.21.0.1 - - [31/Jul/2024:01:16:49 +0000] "GET /console/api/datasets/2826b002-6ad9-40ab-b4ed-e1db06086f40/batch/20240731011039466024/indexing-status HTTP/1.1" 200 335 "http://localhost:82/datasets/7cf35447-72f8-4fec-8bb0-d5e82c4b9111/hitTesting" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/127.0.0.0 Safari/537.36" "-"
nginx-1       | 172.21.0.1 - - [31/Jul/2024:01:16:52 +0000] "GET /console/api/datasets/2826b002-6ad9-40ab-b4ed-e1db06086f40/batch/20240731011039466024/indexing-status HTTP/1.1" 200 335 "http://localhost:82/datasets/7cf35447-72f8-4fec-8bb0-d5e82c4b9111/hitTesting" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/127.0.0.0 Safari/537.36" "-"
worker-1      | Preparing database migration...

**Reply by shanaozi** [reply]
我本地调用站点的向量接口是正常的 但是配置dify就是报错 前段时间还可以这两天都不行,换了很多站点都是一样报错,求求了,哭死

**Reply by L-Timothy** [reply]
你解决了吗 我也遇到这样问题了 哭死

**Comment by cumtsd** [comment]
解决了吗。我也遇到这个问题，无法解决

**Comment by jtsang4** [comment]
Model Provider 那里，不要选择 OpenAI 的分组，选择 OpenAI-API-compatible 这个分组然后填写 embedding 模型就不会报这个错了

**Reply by cumtsd** [reply]
> Model Provider 那里，不要选择 OpenAI 的分组，选择 OpenAI-API-compatible 这个分组然后填写 embedding 模型就不会报这个错了

非常感谢，搞定了。希望不要有其他问题：）

**Reply by LOGIC-10** [reply]
好像全部切换了之后也还是会有这个问题
